packages/transitive-overlay/src/util.ts

```diff
@@ -177,6 +177,27 @@ function makeStop(stop: Place, coordinate?: number[]) {
     : stop;
 }
 
+/**
+ * Helper function that clones the 'to' stop and applies flex logic
+ */
+function cloneToStop(leg: Leg): Place {
+  const toStop = { ...leg.to };
+
+  if (typeof toStop.stop === "object") {
+    toStop.stop = { ...toStop.stop };
+  }
+
+  if (isFlex(leg)) {
+    if (typeof toStop.stop === "object") {
+      toStop.stop.id = `${toStop.stop.id}_flexed_to`;
+    } else {
+      toStop.stopId = `${toStop.stopId}_flexed_to`;
+    }
+  }
```

Comment:
This behavior is not related to stop cloning, and should therefore not be in this function

Reply:
@lightsage88 Rename this function to `makeToStopWithFlexSupport`, and this block will definitely belong here, and that was the intent for extracting this method in the first place. @miles-grant-ibigroup

Reply:
I am renaming it to _makeStopWithFlexSupport_ as a response to this comment: https://github.com/opentripplanner/otp-ui/pull/832#discussion_r2068627210